docs/rest_api.md

```diff
@@ -0,0 +1,289 @@
+# Rest API
+
+We have a basic set of API as of now. All of the services as of now are taking the arguments as parameters and return response.
+
+BMI API assumes that you know the image which you are interested in. BMI has an driver which can communicate with Ceph(the file storage system that we are using for storing our images). Please make sure that you understand Ceph concepts before you play with the API. 
+
+To have a clear understanding of the API, we wish to provide the following terminology:
+* project - A project is the HaaS project that has been allocated to tenant.
+* node - A node that is allocated to project by HaaS. This node should be present in HaaS.
+* img - The name of image which will be used for provisioning.
+* snap_name - The snapshot of the image from which you want to provision a node.
+* network - The provisioning network created in HaaS.
+* nic - The nic which should be used to connect the node to provisioning network
+* channel - The vlan that should be used to connect the node to provisioning network
+
+The convention that we are following for requests is:
+* PUT - for resource creation like node creation 
+* DELETE - for resource deletion for node deletion
+* POST - for rest of operations
+
+**The username and password for HaaS needs to be passed along using HTTP Basic Auth to each possible API call.**
+
+Each possible API call has:
+* an HTTP method and URL path
+* Request body(which will always be form encoded parameters)
+* A list of possible responses
+* An example
+
+---
+###Provision:
+Provision API is needed for provisioning a node from MOC cluster as of now. Provision operation internally calls a ceph clone operation. Ceph clone operation usually takes a snap_
+
+Following is the call for API:
+
+####Link:
+http://BMI_SERVER:PORT/provision/
+
+####Request Type:
+PUT
+
+####Request Body:
+```json
+{
+ "project" : "<project_name>",
+ "node" : "<node_name>" , 
+ "img" : "<image_name>" ,
+ "network" : "<network_name>" ,
+ "channel" : "<channel in haas>" ,
+ "nic" : "<nic to connect on>"
+}
+```
+
+####Responses:
+* 200. This means the provision call is successful.
+* Internal 500 with some junk characters. This means the request body is not proper.
```

Comment:
500 means Internal Server Error. It means the request was proper but the server failed to process it for some fault of the server, not the client or the request.

The correct return code here is [400](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_Error).